model/src/main/java/com/basistech/rosette/dm/Entity.java

```diff
@@ -0,0 +1,283 @@
+ * A reference to a "real world" entity. Each entity in a document is
+ * the result of resolving one or more {@link Mention}s -- a {@linkplain Mention}
+ * is a span of text that mentions an entity, while an {@linkplain Entity}
+ * describes the entity itself.
+ * <p>
+ * Each {@linkplain Entity} provides:
+ * <ul>
+ *     <li>a list of {@linkplain Mention}s</li>
+ *     <li>(optionally) an ID that associates the entity some external knowledge base, e.g.
+ * <a href="http://www.wikidata.org/wiki/Q23">Q23</a> from Wikidata.</li>
+ * <li>(optionally) a sentiment category associated with it
+ * (e.g. "positive", "negative", "neutral").</li>
+ * </ul>
+ * Entities are not spans of text.
+ * The mentions in an entity are in document order.
+ */
+public class Entity extends BaseAttribute {
+    private List<Mention> mentions;
+    private final Integer headMentionIndex;
+    private final String type;
+    private final String entityId;
+    private final Double confidence;
+    private final List<CategorizerResult> sentiment;
+
+    protected Entity(List<Mention> mentions,
+                     Integer headMentionIndex,
+                     String type,
+                     String entityId,
+                     Double confidence,
+                     List<CategorizerResult> sentiment,
+                     Map<String, Object> extendedProperties) {
+        super(extendedProperties);
+        this.mentions = listOrNull(mentions);
+        this.headMentionIndex = headMentionIndex;
+        this.type = type;
+        this.entityId = entityId;
+        this.confidence = confidence;
+        this.sentiment = listOrNull(sentiment);
+    }
+
+    /**
+     * Returns the unique identifier of this entity.
+     *
+     * @return the unique identifier of this entity
+     */
+    public String getEntityId() {
+        return entityId;
+    }
+
+    /**
+     * @return the list of mentions that support this entity.
+     */
+    public List<Mention> getMentions() {
+        return mentions;
+    }
+
+    /**
+     * Return the head mention index, if any. The head mention is the mention judged to be
+     * the best representation of the entity itself. This returns {@code null} if no
+     * mention is designated as head.
+     * @return the index of the head mention.
+     */
+    public Integer getHeadMentionIndex() {
+        return headMentionIndex;
+    }
+
+    /**
+     * Returns the confidence for this resolved entity, or null if there is none.
+     *
+     * @return the confidence for this resolved entity, or null if there is none
+     */
+    public Double getConfidence() {
+        return confidence;
+    }
+
+    /**
+     * Returns the sentiment of this entity, or null if not computed.
+     *
+     * @return the sentiment of this entity, or null if not computed.
+     */
+    public List<CategorizerResult> getSentiment() {
+        return sentiment;
+    }
+
+    /**
+     * If there is a type established for the entity, return the entity.
```

Comment:
typo: "return the type."  (Though I'd probably say something like "Returns the entity type,
or null if none was established.")